### ISSUE TITLE
group_module requires gid with OS X

### ISSUE THREAD

**Issue by anthonygreen** [ansible/ansible-modules-core#1922]
##### Issue Type:

Bug Report
##### Ansible Version:

ansible 1.9.2
##### Environment:

Running from: OS X 10.95
Managing: OS X 10.10.2
##### Summary:

With OSX Ansible group_module requires the gid flag
##### Steps To Reproduce:

Using Ansible to  create a group in OS X

```
- name: Add jenkins group
  group: name=jenkins
  sudo: yes
```

results in a failure      

```
failed: [jenkins_slave_osx] => {"failed": true, "parsed": false}

BECOME-SUCCESS-tevltoiitscjxppvfqwhhbcoujsoocfh
Traceback (most recent call last):
  File "/Users/mediaplayout/.ansible/tmp/ansible-tmp-1439805718.9-51337711497513/group", line 2051, in <module>
    main()
  File "/Users/mediaplayout/.ansible/tmp/ansible-tmp-1439805718.9-51337711497513/group", line 421, in main
    (rc, out, err) = group.group_add(gid=group.gid, system=group.system)
  File "/Users/mediaplayout/.ansible/tmp/ansible-tmp-1439805718.9-51337711497513/group", line 276, in group_add
    (rc, out, err) = self.execute_command(cmd)
  File "/Users/mediaplayout/.ansible/tmp/ansible-tmp-1439805718.9-51337711497513/group", line 96, in execute_command
    return self.module.run_command(cmd)
  File "/Users/mediaplayout/.ansible/tmp/ansible-tmp-1439805718.9-51337711497513/group", line 1895, in run_command
    args = [ os.path.expandvars(os.path.expanduser(x)) for x in args ]
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/posixpath.py", line 261, in expanduser
    if not path.startswith('~'):
AttributeError: 'NoneType' object has no attribute 'startswith'
```

This task does work:

```
- name: Add jenkins group
  group: name=jenkins gid=500 
  sudo: yes
```

**Comment by thudry** [comment]
This should be fixed in the devel
https://github.com/ansible/ansible-modules-core/pull/1086

**Comment by bcoca** [comment]
this is one of the fixes included in 1.9.3
